### PR TITLE
[1.19] Add support for passing certain methods thru to the Agglomerated Trinkets

### PR DESCRIPTION
**Pull request by Dragon-Seeker**
PR Contains ways of using scrolling to select a given trinket and then passing thru certain methods to certain trinket.

Was designed to fix issue with Artifacts that can be disabled never being able to be re-enabled causing it to be useless

### CODE REVIEW

**Comment by Dragon-Seeker** [comment]
Review Anytime Soon?

File: src/main/java/com/glisco/things/client/ThingsClient.java [hunk 1]
```diff
@@ -68,6 +75,31 @@ public void onInitializeClient() {
                 ThingsNetwork.CHANNEL.clientHandle().send(new ThingsNetwork.OpenEnderChestPacket());
             }
         });
+
+        ScreenEvents.AFTER_INIT.register((client, screen, scaledWidth, scaledHeight) -> {
+            if(!(screen instanceof HandledScreen) || !Things.CONFIG.enableAgglomerationInvScrollSelection()) return;
+
+            ScreenMouseEvents.allowMouseScroll(screen).register((screen1, mouseX, mouseY, horizontalAmount, verticalAmount) -> {
+                var slot = ((HandledScreenAccessor) screen1).thing$getSlotAt(mouseX, mouseY);
```
Comment (gliscowo) [inline src/main/java/com/glisco/things/client/ThingsClient.java:83]: `thing` moment

File: src/main/java/com/glisco/things/items/trinkets/AgglomerationItem.java [hunk 2]
```diff
@@ -53,6 +62,80 @@ public AgglomerationItem() {
         super(new Item.Settings().maxCount(1).rarity(Rarity.UNCOMMON));
     }
 
+    //--------
+
+    @Override
+    public boolean onClicked(ItemStack stack, ItemStack otherStack, Slot slot, ClickType clickType, PlayerEntity player, StackReference cursorStackReference) {
```
Comment (gliscowo) [inline src/main/java/com/glisco/things/items/trinkets/AgglomerationItem.java:73]: Generally this implementation is fine, not a large fan of `stack1` though - giving it a more descriptive name like `innerStack` or something similar is preferrable

File: src/main/java/com/glisco/things/items/trinkets/AgglomerationItem.java [hunk 3]
```diff
@@ -53,6 +62,80 @@ public AgglomerationItem() {
...
+    }
+
+    @Override
+    public boolean isUsedOnRelease(ItemStack stack) {
+        return getStackAndRun(stack, null, ItemStack::isUsedOnRelease);
+    }
+
+    //--------
+
+    public <T> T getStackAndRun(ItemStack stack, PlayerEntity player, Function<ItemStack, T> methodPassthru){
```
Comment (gliscowo) [inline src/main/java/com/glisco/things/items/trinkets/AgglomerationItem.java:128]: `"SelectedTrinket"` should really be an NbtKey

**Review by gliscowo — CHANGES_REQUESTED** [review]

File: src/main/java/com/glisco/things/client/ThingsClient.java (hunk 1 above)
Comment (Dragon-Seeker) [reply src/main/java/com/glisco/things/client/ThingsClient.java:83]: yep

**Pull request merged**